modules/core/src/main/scala/org/tessellation/l0/infrastructure/snapshot/SnapshotService.scala

```diff
@@ -2,46 +2,82 @@ package org.tessellation.l0.infrastructure.snapshot
 
 import cats.data.NonEmptySet
 import cats.effect.kernel.Async
+import cats.effect.std.Queue
+import cats.syntax.flatMap._
 import cats.syntax.functor._
+import cats.syntax.semigroup._
 
+import scala.concurrent.duration.DurationInt
 import scala.util.control.NoStackTrace
 
 import org.tessellation.dag.domain.block.DAGBlock
-import org.tessellation.kernel.StateChannelSnapshot
-import org.tessellation.l0.domain.snapshot.{GlobalSnapshot, SnapshotService, SnapshotStorage}
+import org.tessellation.ext.fs2.switchRepeat
+import org.tessellation.l0.domain.snapshot._
 import org.tessellation.schema.height.Height
 import org.tessellation.schema.peer.PeerId
-import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
 
-import eu.timepit.refined.auto._
-import eu.timepit.refined.types.numeric.PosLong
+import eu.timepit.refined.cats.nonNegLongMonoid
+import eu.timepit.refined.types.numeric.NonNegLong
+import fs2.Stream
 import io.estatico.newtype.ops._
 
 object SnapshotService {
 
-  private val heightInterval = PosLong(2L)
-
   def make[F[_]: Async](
-    snapshotStorage: SnapshotStorage[F]
+    snapshotStorage: SnapshotStorage[F],
+    blocksQueue: Queue[F, Set[Signed[DAGBlock]]]
   ): SnapshotService[F] =
     new SnapshotService[F] {
 
-      def createSnapshot(
-        blocks: NonEmptySet[Hashed[DAGBlock]],
-        snapshots: Set[StateChannelSnapshot],
-        nextFacilitators: NonEmptySet[PeerId]
-      ): F[GlobalSnapshot] =
+      def offer(signedBlock: Signed[DAGBlock]): F[Unit] =
         for {
-          previousSnapshotHeight <- snapshotStorage.getLastSnapshotHeight
-          nextSnapshotHeight = Height(previousSnapshotHeight.coerce + heightInterval)
+          _ <- blocksQueue.offer(Set(signedBlock))
+        } yield ()
+
+      private val emptySnapshot = Stream.eval(createEmptySnapshot())
+
+      private val normalSnapshot = Stream.fromQueueUnterminated(blocksQueue).evalMap(createNormalSnapshot)
+
+      val snapshotCreation: Stream[F, GlobalSnapshot] =
+        normalSnapshot.through(switchRepeat(10.seconds, emptySnapshot))
```

Comment:
please extract delays into one place (config preferable)

Reply:
I can extract that but the question is if It can be configurable by the node operator

Reply:
We can have a config class or even object with meaningful names that's not configurable. If it's not configurable then maybe let's call it differently, I don't know parameters?

Reply:
@TheMMaciek we already have non-configurable config